### PR TITLE
implement function factory

### PR DESCRIPTION
**Pull request by hootkoMan**

### CODE REVIEW

File: function-factory-task/src/main/java/com/bobocode/FunctionFactoryTask.java [hunk 1]
```diff
@@ -42,24 +42,32 @@ public InvalidFunctionNameException(String functionName) {
     public static void main(String[] args) {
         FunctionFactory<Integer, Integer> functionFactory = new FunctionFactory<>();
         // 1. add simple functions "square", "increment", "decrement", "negative"
+        Function<Integer, Integer> fSquare = f -> f * f;
+        functionFactory.addFunction("square", fSquare);
+        functionFactory.addFunction("increment", f -> f + 1);
+        functionFactory.addFunction("increment2", f-> { f++; int i = f; return  i;});
```
Comment (tboychuk) [inline function-factory-task/src/main/java/com/bobocode/FunctionFactoryTask.java:48]: Seriously? :)

**Review by tboychuk — COMMENTED** [review]
Good 👍

File: function-factory-task/src/main/java/com/bobocode/FunctionFactoryTask.java (hunk 1 above)
Comment (hootkoMan) [reply function-factory-task/src/main/java/com/bobocode/FunctionFactoryTask.java:48]: Yes)) I tested post increment and it work !

Comment (tboychuk) [reply function-factory-task/src/main/java/com/bobocode/FunctionFactoryTask.java:48]: I mean why did you do that? ))

**Pull request closed without merging**